### PR TITLE
`ModelAnalysis.create(...)`

### PR DESCRIPTION
**Pull request by rahul-tuli**
This PR adds a `.create(...)` function for `ModelAnalysis`, this makes it slightly easier to run `ModelAnalysis` on a variety of input:

     - [X] onnx models
     - [X] deployment directory
     - [X] SparseZoo Stubs
     - [X] yaml files and raw yaml strings

All of the above can now be used to create a ModelAnalysis, for example:

```python
from sparsezoo.analyze import ModelAnalysis

path_or_stub = ... # SparseZoo stub or onnx model or deployment dir path or yaml file or raw yaml string
ModelAnalysis.create(path_or_stub)
```

Before this PR doing the same as above would require a cumbersome `if-else` ladder + error handling. that had the potential to be duplicated all across the codebase; this change now promotes one-source of truth for managing supported inputs

### CODE REVIEW

File: src/sparsezoo/analyze/analysis.py [hunk 1]
```diff
@@ -964,6 +966,45 @@ def from_onnx(cls, onnx_file_path: Union[str, ModelProto]):
...
+
+        if isinstance(file_path, ModelProto) or Path(file_path).suffix == ".onnx":
+            return ModelAnalysis.from_onnx(onnx_file_path=file_path)
+
+        if file_path.startswith("zoo:"):
+            return ModelAnalysis.from_onnx(
+                Model(file_path).deployment.get_file("model.onnx").path
+            )
+        if Path(file_path).is_dir():
+            return ModelAnalysis.from_onnx(Path(file_path) / "model.onnx")
```
Comment (bfineran) [inline src/sparsezoo/analyze/analysis.py:996]: let's add a log to this one

**Review by bfineran — DISMISSED** [review]
lgtm pending minor comments

File: src/sparsezoo/analyze/analysis.py [hunk 2]
```diff
@@ -964,6 +966,45 @@ def from_onnx(cls, onnx_file_path: Union[str, ModelProto]):
...
+
+        if Path(file_path).is_file() and Path(file_path).suffix == ".yaml":
+            return ModelAnalysis.parse_yaml_file(file_path=file_path)
+
+        if isinstance(file_path, str):
```
[the remark this answers is not in the document]
Comment (rahul-tuli) [reply src/sparsezoo/analyze/analysis.py:1001]: Added this check in the first if, good catch

[the remark this answers is not in the document]
Comment (bfineran) [reply src/sparsezoo/analyze/analysis.py:1007]: we can keep both, just if it is a file, load onnx regardless of extension

**Review by bfineran — APPROVED** [review]

File: src/sparsezoo/analyze/analysis.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rahul-tuli) [reply src/sparsezoo/analyze/analysis.py:1009]: Done!

**Review by dbogunowicz — APPROVED** [review]

**Pull request merged**